packages/application-shell/src/components/application-shell/application-shell.tsx

```diff
@@ -535,11 +71,11 @@ const ApplicationShell = (props: TApplicationShellProps) => {
           if (isAuthenticated) {
             return (
               <Switch>
-                <Route path="/logout">
+                <SuspendedRoute path="/logout">
                   <RedirectToLogout />
-                </Route>
-                <Route>
-                  <RestrictedApplication
+                </SuspendedRoute>
+                <SuspendedRoute>
+                  <ApplicationShellAuthenticated
```

Comment:
In general I think this is a great idea.

However, I would also share some thoughts about lazy loading component with a very high chance of been rendered as this `ApplicationShellAuthenticated`, or project and locale switchers. By moving them out from the initial bundle, we are introducing a delay in the user experience since now they will need to be fetched when React detects it needs them.

Maybe the latency for a backoffice application is not such a high requirement because most users will have a wired internet connection, but just wanted to share some concerns I had in the past when using lazy loading too much.

Reply:
Thanks for the feedback. Yes some parts might have a slight delay now. The `<ApplicationShellAuthenticated>` is only rendered if the user is authenticated so I think we can try code splitting it.

In general we can observe the loading behavior / performance and make further adjustments along the way.

Regarding the switcher components, you're right they probably don't need to be code splitted. Let me revert that part.